### ISSUE TITLE
CanWrite passes if any child node can be written

### ISSUE THREAD

**Issue by rayd** [goldibex/targaryen#9]
Hi, I'm seeing tests pass that I don't think should actually be passing. I have a set of rules that look something like this (simplified a bit here):
```
{
  "rules":{
    ".write":"false",
    ".read":"false",
    "accounts": {
      ".write":"false",
      ".read":"true",
      "$accountId": {
        ".write":"false",
        ".read":"true",
        "flows": {
          ".write":"false",
          ".read":"true",
          "$flowId": {
            ".write":"false",
            "name": {
              ".write":"true",
              ".read":"true"
            },
            "createdAt": {
              ".write":"true",
              ".read":"true"
            },
            "steps": {
              ".write":"false",
              ".read":"true",
              "$stepId": {
                ".write":"true",
                ".read":"true",
                "stepType": {
                  ".write":"true",
                  ".read":"true"
                },
                "content": {
                  ".write":"true",
                  ".read":"true"
                }
              }
            }
          }
        }
      }
    }
  }
}
```
If I attempt to write the following data, it passes the write test (which it shouldn't according to my understanding of the rules):
```
  "tests": {
    "accounts/68/flows" : {
      "canWrite" : [{
        "auth": "User2000",
        "data": {
          "-Jkaaiir8329": {
            "name": "Test Flow",
            "createdAt": 123431411,
            "steps": {
              "-J9847saf7fhu": {
                "stepType": "image",
                "content": "test"
              }
            }
          }
        }
      }]
    }
  }
```
The rule should fail at this part I would think:
```
"$flowId": {
    ".write":"false",
```
But instead it looks like targaryen keeps evaluating child rules, finding one that passes. Here's the debug output, showing why the test passes:
```
/:.write: "false"
    => false
/accounts:.write: "false"
    => false
/accounts/68:.write: "false"
    => false
/accounts/68/flows:.write: "false"
    => false
/accounts/68/flows/-Jkaaiir8329:.write: "false"
    => false
/accounts/68/flows/-Jkaaiir8329/name:.write: "true"
    => true
Write was allowed.

┌───────────────────┬───────┬──────────┬────────┬─────┐
│ Path              │ Type  │ Auth     │ Expect │ Got │
├───────────────────┼───────┼──────────┼────────┼─────┤
│ accounts/68/flows │ write │ User2000 │ ✓      │ ✓   │
└───────────────────┴───────┴──────────┴────────┴─────┘
```
Maybe I'm using this incorrectly or have an incorrect assumption about how it should be working, but I think this may be a bug?

Thanks for the awesome tool by the way -- super helpful.

**Comment by goldibex** [comment]
There's a lot of redundancy in your rules. They can be simplified to the following with no change in behavior:

```json
{
  "rules":{
    "accounts": {
      ".read":"true",
      "$accountId": {
        "flows": {
          "$flowId": {
            "name": {
              ".write":"true"
            },
            "createdAt": {
              ".write":"true"
            },
            "steps": {
              "$stepId": {
                ".write":"true"
              }
            }
          }
        }
      }
    }
  }
}
```

That said, it does appear some bug is cropping up here. Thanks for the report, I'll issue a fix in pretty short order (EOW latest).

**Comment by goldibex** [comment]
@rayd: can you upgrade to Targaryen 2.1.0 and see if your problem persists?

**Comment by goldibex** [comment]
I think this was closed in #27.
